lib/common/client/concerns/mhv_locked_session_client.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Concerns
+      ##
+      # Module mixin for overriding session logic when making session-based client connections that
+      # should lock during session creation, to prevent threads from making simultaneous
+      # authentication API calls.
+      #
+      # All refrences to "session" in this module refer to the upstream MHV session.
+      #
+      # @see MedicalRecords::Client
+      #
+      # @!attribute [r] session
+      #   @return [Hash] a hash containing session information
+      #
+      module MhvLockedSessionClient
+        extend ActiveSupport::Concern
+        include SentryLogging
+
+        LOCK_RETRY_DELAY = 1 # Number of seconds to wait between attempts to acquire a session lock
```

Comment:
Could we lower this value? Say to 0.1 with an exponential back-off?

Would there be a reason not to do this? I don't like that `sleep` blocks the thread here for so long.

Reply:
@ericboehs I was also concerned about the sleep, but in my tests, it typically takes about 2-4 seconds to create a session. This may be somewhat faster on the server, but because of the remote requests, it is definitely in the multi-second range, as opposed to sub-second range.

I can definitely implement an exponential backoff capped at 1 second, but I don't necessarily foresee a circumstance where it locks for much less than 1 second. What do you think?